neo4j/db/errors.go

```diff
@@ -73,6 +74,18 @@ func (e *Neo4jError) parse() {
 	e.title = parts[3]
 }
 
+// reclassify classifies specific errors coming from pre-5.x servers into their
+// 5.x classifications
+// this function can be removed once support for pre-5.x servers is dropped
+func (e *Neo4jError) reclassify() {
+	switch e.Code {
+	case "Neo.TransientError.Transaction.LockClientStopped":
+		e.Code = "Neo.ClientError.Transaction.LockClientStopped"
+	case "Neo.TransientError.Transaction.Terminated":
+		e.Code = "Neo.ClientError.Transaction.Terminated"
+	}
+}
+
```

Comment:
Why should we reclassify the error in the driver code?

Reply:
I also think this should be done on the server side but that's water under the bridge.
The decision is that drivers are responsible for mapping 4.x errors into their 5.x classifications since retries are a driver-only concern.